src/dune_digest/digest.ml

```diff
@@ -91,11 +91,13 @@ let file_with_executable_bit ~executable path =
 module Stats_for_digest = struct
   type t =
     { st_kind : Unix.file_kind
-    ; st_perm : Unix.file_perm
+    ; executable : bool
     }
 
   let of_unix_stats (stats : Unix.stats) =
-    { st_kind = stats.st_kind; st_perm = stats.st_perm }
+    (* Check if any of the +x bits are set, ignore read and write *)
+    let executable = 0o111 land stats.st_perm <> 0 in
+    { st_kind = stats.st_kind; executable }
```

Comment:
I kinda wonder whether `Path.Permissions.executable` would make sense to be used here, but maybe it is not worth the hassle.

Reply:
`Path.Permissions.(test executable)` will only test if the flag is set for the current user (u+x, or `0o100`)
I wanted here to explicitly test if that flag is set for anyone.
Granted, files with [u-x g+x o+x] are probably few and far between, but still.